### ISSUE TITLE
Create `Certificate` model

### ISSUE THREAD

**Issue by dickdavis** [dickdavis/oauth-flow-demo#56]
Add a `Certificate` model that represents client configuration records stored on the `certificates` table. The model should:
* validate issuer, presence true
* validate subject, presence true
* validate serial, presence true
* validate not before, presence true
* validate not after, presence true
* validate that the certificate is not self-signed (can be implemented in a follow-up effort)
* validate that the certificate is not expired (can be implemented in a follow-up effort)

The model should also provide a `from_plaintext` class method that reads a plaintext certificate, parses the attributes, and saves the record.

The model should also have a belongs to association with a client. The `Client` model should also be updated to have a has many relationship with certificates.

The model should be read-only after it has been created.
